src/libraries/System.Net.WebSockets.Client/tests/ConnectTest.Http2.cs

```diff
@@ -96,6 +96,59 @@ public async Task ConnectAsync_VersionNotSupported_WithSsl_Throws()
             );
         }
 
+        [OuterLoop("Uses external servers", typeof(PlatformDetection), nameof(PlatformDetection.LocalEchoServerIsNotAvailable))]
+        [Theory]
+        [MemberData(nameof(SecureEchoServersAndBoolean))]
+        [SkipOnPlatform(TestPlatforms.Browser, "System.Net.Sockets is not supported on this platform")]
+        public async Task ConnectAsync_Http11Server_DowngradeFail(Uri server, bool useHandler)
+        {
+            using (var cws = new ClientWebSocket())
+            using (var cts = new CancellationTokenSource(TimeOutMilliseconds))
+            {
+                cws.Options.HttpVersion = HttpVersion.Version20;
+                cws.Options.HttpVersionPolicy = Http.HttpVersionPolicy.RequestVersionExact;
+                Task t;
+                if (useHandler)
+                {
+                    var handler = new SocketsHttpHandler();
+                    t = cws.ConnectAsync(server, new HttpMessageInvoker(handler), cts.Token);
+                }
+                else
+                {
+                    t = cws.ConnectAsync(server, cts.Token);
+                }
+                var ex = await Assert.ThrowsAnyAsync<WebSocketException>(() => t);
+                Assert.IsType<HttpRequestException>(ex.InnerException);
+                Assert.True(ex.InnerException.Data.Contains("HTTP2_ENABLED"));
+                Assert.Equal(WebSocketState.Closed, cws.State);
+            }
+        }
+
+        [OuterLoop("Uses external servers", typeof(PlatformDetection), nameof(PlatformDetection.LocalEchoServerIsNotAvailable))]
+        [Theory]
+        [MemberData(nameof(EchoServersAndBoolean))]
```

Comment:
Do EchoServersAndBoolean include secure server as well?

Reply:
Yes, it includes both